test/core.jl

```diff
@@ -6915,3 +6915,17 @@ TupleOf31406(cols::Union{Shape31406,Type}...) = TupleOf31406(collect(Shape31406,
         end
         true
     end
+
+# Issue #31649
+struct sparse_t31649
+    val::Vector{Float64}
+    sub::Vector{Int64}
+end
+Base.convert(::Any, v::sparse_t31649) = copy(v.val)
```

Comment:
Maybe adding this method is asking for trouble?

Reply:
I considered it. On the other hand, asking for trouble is kind of the point.